woocommerce/woocommerce
# Update plugins installer component to TS, Fix TS bugs and Syncpack TypeScript

### All Submissions:






### Changes proposed in this Pull Request:

After upgrading to pnpm 7, intermittent errors were occuring on build:

```
woocommerce/client/admin:turbo:build: ERROR in client/tasks/fills/tax/woocommerce-tax/plugins.tsx:77:5
woocommerce/client/admin:turbo:build: TS2322: Type '{ onComplete: (activatedPlugins: string[], response: InstallPluginsResponse) => void; onError: (errors: unknown, response: unknown) => void; onSkip: () => void; skipText: string; onAbort: () => void; abortText: string; }' is not assignable to type 'IntrinsicAttributes & { children?: ReactNode; }'.
woocommerce/client/admin:turbo:build:   Property 'onComplete' does not exist on type 'IntrinsicAttributes & { children?: ReactNode; }'.
```

This PR updates the `Plugins` installer component to a functional TS component with respective types to fix the issue. It also adjusts the tests to match the fact that we can't mock `installAndActivatePlugins` in the same way now its declared inside the component.

Additionally this fixes some other small TS issues that arose and matches TypeScript version across repo with Syncpack so that every project is compiled with the same version.

### How to test the changes in this Pull Request:

1. Check out this branch
2. Run `pnpm i && pnpm run build`
3. Smoke test all uses of the plugin component (taxes task, experimental shipping task, payments task, WCPay task, etc)
4. Check that the plugins are install/activated correctly
5. Deactivate plugins and attempt other actions in the component, such as skipping (note that skip text may be different in each component)

### Other information:




### FOR PR REVIEWER ONLY:
